### PR TITLE
Remove the udp_fallback function

### PR DESCRIPTION
**Pull request by thsfs**
- Remove the udp_fallback function.
- Add complementary log.

Fixes issue: https://github.com/nanocurrency/nano-node/issues/3846

### CODE REVIEW

**Review by clemahieu — DISMISSED** [review]
LGTM

**Comment by thsfs** [comment]
> LGTM

Sorry, there was a `!ec` in use in one of the try_log().Removed it.

File: nano/node/transport/tcp.cpp [hunk 1]
```diff
@@ -533,11 +533,7 @@ void nano::transport::tcp_channels::update (nano::tcp_endpoint const & endpoint_
 
 void nano::transport::tcp_channels::start_tcp (nano::endpoint const & endpoint_a)
 {
-	if (node.flags.disable_tcp_realtime)
-	{
-		node.network.tcp_channels.udp_fallback (endpoint_a);
-		return;
-	}
+	debug_assert (!node.flags.disable_tcp_realtime);
```
Comment (dsiganos) [inline nano/node/transport/tcp.cpp]: Is it certain that disable_tcp_realtime cannot be true?

**Review by dsiganos — DISMISSED** [review]

File: nano/node/transport/tcp.cpp (hunk 1 above)
Comment (thsfs) [reply nano/node/transport/tcp.cpp]: The opposite of it was the precondition for the udp fallback to be used. You are right, we cannot assume this is negative after removing udp. Will remove it.

Comment (dsiganos) [reply nano/node/transport/tcp.cpp]: But now, there is no check for disable_tcp_realtime. Either we should have an if statement like before or an assert. Now we have none of the 2.

**Pull request merged**